python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1734,6 +1779,19 @@ def create_reexecuted_run(
                 parent_run=parent_run,
             )
             tags[RESUME_RETRY_TAG] = "true"
+        elif strategy == ReexecutionStrategy.FROM_ASSET_FAILURE:
+            skipped_asset_keys, skipped_asset_check_keys = self._get_skipped_entity_keys(
+                parent_run_id
+            )
+            remote_job = code_location.get_job(
```

Comment:
doesn't seem like we do a good job of handling this in general; but should we be gracefully failing if this thing doesn't exist anymore? It might not right, if the job has been removed from code location, or if the set of asset keys has changed?

Reply:
my thinking is that the other retry from failure codepath would also fail if the underlying job changed -- a hard error doesn't seem too bad? although I'll admit to not really knowing how that manifests in the UI